csharp/Adapter/Microsoft.Spark.CSharp/Configuration/ConfigurationService.cs

```diff
@@ -150,14 +150,25 @@ internal SparkCLRLocalConfiguration(System.Configuration.Configuration configura
 
             internal override string GetCSharpRDDExternalProcessName()
             {
-                return appSettings.Settings["CSharpWorkerPath"].Value;
+                KeyValueConfigurationElement workerPathConfig = appSettings.Settings["CSharpWorkerPath"];
+                if (workerPathConfig != null)
+                {
+                    // Use the file name part of the CSharpWorkerPath value, if specified
+                    return Path.GetFileName(workerPathConfig.Value);
```

Comment:
For SparkCLRLocalConfiguration, CSharpWorkerPath is a required parameter in appSettings. If it is not set, we should just throw exception. Using base.GetCSharpRDDExternalProcessName value would lead to exception later.